Minio.Functional.Tests/FunctionalTest.cs

```diff
@@ -2393,6 +2395,133 @@ internal static async Task ObjectRetentionAsync_Test1(MinioClient minio)
 
     #endregion
 
+    internal static MemoryStream CreateZipFile(string prefix, int nFiles)
+    {
+        var outputMemStream = new MemoryStream();
+        var zipStream = new ZipOutputStream(outputMemStream);
+
+        zipStream.SetLevel(3); //0-9, 9 being the highest level of compression
+        byte[] bytes = null;
+
+        for (var i = 0; i <= nFiles; i++)
+        {
+            // Make one large, compressible file.
+            if (i == nFiles) i = 1000000;
+
+            var fileName = prefix + "file-" + i + ".bin";
+            Directory.CreateDirectory(prefix);
+            var newEntry = new ZipEntry(fileName);
+            newEntry.DateTime = DateTime.Now;
+
+            zipStream.PutNextEntry(newEntry);
+
+            bytes = rsg.GenerateStreamFromSeed(i).ToArray();
+
+            var inStream = new MemoryStream(bytes);
+            if (i == 0) StreamUtils.Copy(inStream, zipStream, new byte[128]);
+            else StreamUtils.Copy(inStream, zipStream, new byte[i * 128]);
+
+            inStream.Close();
+            zipStream.CloseEntry();
+        }
+
+        zipStream.IsStreamOwner = false; // False stops the Close also Closing the underlying stream.
+        zipStream.Close(); // Must finish the ZipOutputStream before using outputMemStream.
+
+        outputMemStream.Position = 0;
+
+        outputMemStream.Seek(0, SeekOrigin.Begin);
+
+        return outputMemStream;
+    }
+
+    internal static async Task GetObjectS3Zip_Test1(MinioClient minio)
```

Comment:
Is this test run in CI? How do I verify it was run?

Reply:
Yes, it does. As part of the build workflow testing.
It is added into the list of tests run in `Minio.Functional.Tests/Program.cs` file, line #169